File: Backend/Microservices/Finance/src/main/java/tn/esprit/finance/entity/Payment.java
package tn.esprit.finance.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long payId;

    @ManyToOne
    private Invoice invoiceId;
    private double montantPaye;
    private LocalDateTime datePaiement;

    @Enumerated(EnumType.STRING)
    private PaymentMode modePaiement;

    public enum PaymentMode{
        VIREMENT,
        CHEQUE,
        CARTE_BANCAIRE,
        ESPECE
    }
}

File: Frontend/FrontOffice_ConstructionIQ/src/app/models/payment.ts
export class Payment {
  id: number;
  commandeId: number;  // L'ID de la commande associée
  invoiceId: number;   // L'ID de la facture associée
  montantPaye: number;  // Montant payé
  datePaiement: string;  // Date du paiement
  modePaiement: PaymentMode;

  constructor(id: number = 0, commandeId: number = 0, invoiceId: number = 0, montantPaye: number = 0, datePaiement: string = '', modePaiement: PaymentMode = PaymentMode.VIREMENT) {
    this.id = id;
    this.commandeId = commandeId;
    this.invoiceId = invoiceId;
    this.montantPaye = montantPaye;
    this.datePaiement = datePaiement;
    this.modePaiement = modePaiement;
  }
}

export enum PaymentMode {
  VIREMENT = 'VIREMENT',
  CHEQUE = 'CHEQUE',
  CARTE_BANCAIRE = 'CARTE_BANCAIRE'
}
